### ISSUE TITLE
Locked pages report non distinct 'locked by' user filter resulting in fatal error when applied

### ISSUE THREAD

**Issue by richie-blake** [wagtail/wagtail#10960]
### Issue Summary

On the locked pages report, the dropdown for the 'locked by' filter is populated with a non distinct list of users. i.e. if a user has locked more than one page, their username will appear in the 'locked by' filter for every page they have locked. 
<img width="431" alt="Screen Shot 2023-09-28 at 11 41 45 AM" src="https://github.com/wagtail/wagtail/assets/4099711/473f52c5-a806-41e5-b1c8-77c39e4097de">

This is caused by the filter queryset code in the "get_users_for_filter" function

```
def get_users_for_filter():
    User = get_user_model()
    return User.objects.filter(locked_pages__isnull=False).order_by(User.USERNAME_FIELD)
```

Applying the filter with this non distinct list results in a fatal error.

MultipleObjectsReturned at /admin/reports/locked/
get() returned more than one User -- it returned 2!

### Steps to Reproduce

1. As an authenticated admin user lock at least two pages in the CMS
2. Browse to Reports -> Locked pages
3. You should see your username listed at least twice in the locked by filter.
4. Select one of those usernames from the list and click 'Apply filter'

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: no

### Technical details

-   Python version: 3.8.x
-   Django version: 3.2.x
-   Wagtail version: Confirmed on 4.1.6 and 4.2.4

### Workaround

Overriding the url route with a custom view and then overriding the filter to return a distinct queryset resolves the issue as a workaround

```
def get_users_for_filter():
    """
    Customisation applied here to make sure this queryset is returned distinct. In the underlying Wagtail
    implementation an undistinct queryset is returned which causes fatal errors when applying filters.
    """
    User = get_user_model()
    return User.objects.filter(locked_pages__isnull=False).order_by(User.USERNAME_FIELD).distinct()

class CustomLockedPagesReportFilterSet(LockedPagesReportFilterSet):
    locked_by = django_filters.ModelChoiceFilter(
        field_name="locked_by", queryset=lambda request: get_users_for_filter()
    )

class CustomLockedPagesView(LockedPagesView):
    filterset_class = CustomLockedPagesReportFilterSet
```

**Comment by rohitsrma** [comment]
I can confirm that the issue exists. When I lock more than one page on the BakeryDemo site, my username appears multiple times in the `Locked by` filter. A screenshot for reference is attached below.

![Screenshot (35)](https://github.com/wagtail/wagtail/assets/111359305/e6dc6fee-9330-48cb-aa3b-c12ec53f5de7)

When selecting any of my usernames and applying the filter, it results in an error saying 
```
MultipleObjectsReturned at /admin/reports/locked/
get() returned more than one User -- it returned 2!
```

Additionally, I would like to work on this issue.

**Comment by rohitsrma** [comment]
As suggested by @richie-blake in the issue, a simple solution to this would be updating the filter to return a distinct queryset.

I've have tested it, and it appears to work well. However, input from other members would be highly valuable. I'd appreciate any suggestions or insights on whether I should proceed with this approach or consider alternative solutions.

**Comment by gasman** [comment]
Thanks for the report @richie-blake! PR now opened at #11245.

@rohitsrma Sorry for jumping ahead of you on this one - I'm keen to get this fix into the 5.2.1 release today/tomorrow, so I wanted to get this sorted quickly. Thanks for investigating!
